### PR TITLE
feature(timeframe selector): add time frame selector for desktop and mobile PE-1300

### PR DESCRIPTION
**Pull request by karlprieb**
https://user-images.githubusercontent.com/9200155/166001926-51f84902-004e-481f-9072-84a1222d7703.mp4

### CODE REVIEW

File: contexts/timeframe_selector.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Dispatch, useState, createContext, SetStateAction, ReactNode } from 'react';
+
+export type TimeframeSelectorOptions = 'weekly' | 'lastWeek' | 'total';
+type TimeframeSelctorContextType = [TimeframeSelectorOptions, Dispatch<SetStateAction<TimeframeSelectorOptions>>];
```
Comment (fedellen) [inline contexts/timeframe_selector.tsx:4]:
```suggestion
type TimeframeSelectorContextType = [TimeframeSelectorOptions, Dispatch<SetStateAction<TimeframeSelectorOptions>>];
```

File: components/header/index.tsx [hunk 2]
```diff
@@ -24,6 +31,7 @@ const Header = (): JSX.Element => {
 			</HeaderWrapper>
 			<LeaderboardWrapper as='header'>
 				<Title>Leaderboard</Title>
+				<TimeframeSelectorDesktop />
```
Comment (fedellen) [inline components/header/index.tsx:34]: Does this component get rendered on Mobile? I'm surprised to see these in separate places

File: components/header/timeframe_selector_mobile.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+				</TimeframeButton>
+				<TimeframeButton selected={timeframe === 'lastWeek'} onClick={() => selectTimeframe('lastWeek')}>
+					{TimeframeNames['lastWeek']}
+					<Checkmark />
+				</TimeframeButton>
+				<TimeframeButton selected={timeframe === 'total'} onClick={() => selectTimeframe('total')}>
+					{TimeframeNames['total']}
+					<Checkmark />
+				</TimeframeButton>
+			</TimeframeButtonWrapper>
```
Comment (fedellen) [inline components/header/timeframe_selector_mobile.tsx:90]: Seems to me that there could be a `Timeframe` component with a `isMobile` prop to conditionally render the checkmarks. Unless I'm missing some other differences here

File: components/header/index.tsx (hunk 2 above)
Comment (karlprieb) [reply components/header/index.tsx:34]: It doesn't get rendered on mobile.
The issue is that on desktop the timeframe selector will be displayed on the header, but on mobile it will be displayed on the main.

File: components/header/timeframe_selector_mobile.tsx (hunk 3 above)
Comment (karlprieb) [reply components/header/timeframe_selector_mobile.tsx:90]: You're right :)

Comment (fedellen) [reply components/header/timeframe_selector_mobile.tsx:90]: The CSS differences are notable though. It would require more media checks to merge it all

File: components/header/index.tsx (hunk 2 above)
Comment (fedellen) [reply components/header/index.tsx:34]: It seems to become hidden, but I do see it in the HTML stack and with React dev tools still. As opposed to the mobile selector modal which disappears from the HTML when closed

Don't worry too much about it for now, its a small performance boost at best

File: components/header/wallet_modal.tsx [hunk 4]
```diff
@@ -75,7 +73,7 @@ const WalletModal = ({
 				</ModalButton>
 				{isArconnectAvailable ? (
 					<ModalButton onClick={() => connectToArconnect()}>
-						<Image alt='ArConnect Logo' src={arconnectLogo} width={22} height={22} />
+						<img alt='ArConnect Logo' src='/arconnect.png' width={22} height={22} />
```
Comment (fedellen) [inline components/header/wallet_modal.tsx:76]: Why the change from `Image` here? Doesn't next.js provide some nice image scaling for us here?

File: components/sidebar/index.tsx [hunk 5]
```diff
@@ -174,7 +182,7 @@ const Sidebar = ({ data }: { data: Data }): JSX.Element => {
 				},
 			}));
 		}
-	}, [walletAddress, data.wallets]);
+	}, [walletAddress, data.wallets, timeframe]);
```
Comment (fedellen) [inline components/sidebar/index.tsx:185]: 👏

Comment (fedellen) [reply components/sidebar/index.tsx:185]: Took me a minute here but I realized that we certainly would want to run this whenever each of these changes

**Review by fedellen — APPROVED** [review]
